crates/sui-core/src/authority_batch.rs

```diff
@@ -350,15 +357,19 @@ impl crate::authority::AuthorityState {
                 if local_state.pending_batch.is_some()
                     && local_state.next_item == NextItemToPublish::Batch
                 {
-                    // We already publihsed all txns for this pending batch, now is the time to
+                    // We already published all txns for this pending batch, now is the time to
                     // publish the batch
                     let batch = local_state.pending_batch.unwrap();
                     local_state.pending_batch = None;
+                    local_state.metrics.follower_batches_streamed.inc();
                     return Some((
                         Ok(BatchInfoResponseItem(UpdateItem::Batch(batch))),
                         local_state,
                     ));
                 } else if local_state.next_item == NextItemToPublish::Transaction {
+                    // TODO: (?) we could have read the entire set of transactions in this batch up to `end`.
+                    // But assuming the recent transactions are very hot and hopefully they can be found in
+                    // rocksdb cache.
```

Comment:
@sadhansood , is `seq_num` in `executed_sequence` still guaranteed to be consecutive? It's very attractive for me to update this `iter().skip_to()` to `get()` so we can leverage point lookup cache more

Reply:
@longbowlu if you are worried about performance, we could potentially cache the txns in memory for a batch in `local_state` given one batch is never going to have more than 10 txns and then stream it instead of pointing the iterator to next tx every time. I think it is probably better to not assume anything about the consecutive sequence of the txns here as that is better left for the batch maker to decide (and is free to change the implementation in future), otherwise we might be coupling the two components tightly.